great_tables/_locations.py

```diff
@@ -417,11 +566,46 @@ def _(loc: LocBody, data: GTData, style: list[CellStyle]) -> GTData:
     for col_pos in positions:
         row_styles = [entry._from_row(data._tbl_data, col_pos.row) for entry in style_ready]
         crnt_info = StyleInfo(
-            locname="data", locnum=5, colname=col_pos.colname, rownum=col_pos.row, styles=row_styles
+            locname="cell", locnum=5, colname=col_pos.colname, rownum=col_pos.row, styles=row_styles
         )
         all_info.append(crnt_info)
 
-    return data._replace(_styles=data._styles + all_info)
+    data._styles.extend(all_info)
```

Comment:
I think we may need to convert every use of `data._styles.extend(...)` or `data._styles.append(...)` to `data._replace(...)`.

The extend and append methods mutate the styles list, which can cause style changes to apply to unanticipated places. (`._replace()` does a shallow copy so should be fast to do)

Reply:
Ok I went the other way, but I'll revert back!